Client/macos/MacOSBluetoothConnector.mm

```diff
@@ -63,29 +63,42 @@ -(void)rfcommChannelData:(IOBluetoothRFCOMMChannel *)rfcommChannel data:(void *)
     asyncCommDelegate->delegateCPP = macOSBluetoothConnector;
     // try to open channel
     if ( [device openRFCOMMChannelAsync:&channel withChannelID:rfcommChannelID delegate:asyncCommDelegate] != kIOReturnSuccess ) {
-        throw RecoverableException("Error - could not open the rfcomm.\n", true);
+        RecoverableException exc = RecoverableException("Could not open the rfcomm.", false);
+        std::exception_ptr excPtr = std::make_exception_ptr(exc);
```

Comment:
when reading the std::make_exception_ptr docs I was kinda terrified for a minute, because I thought that it slices everything to an std::exception 😅